admin-src/index.js

```diff
@@ -1,111 +1,37 @@
-
-          await Promise.all(
-            querySnapshot.docChanges().map(async docChange => {
-              const doc = docChange.doc
-              const changeType = docChange.type
-
-              if (changeType === 'added') {
-                if (!unanswered.find(diagnose => diagnose.id === doc.id)) {
-                  const docData = doc.data()
-                  const userSnap = await firebase.firestore().collection('users').doc(docData.user).get()
-                  const username = userSnap.data().username
-                  unanswered.push({
-                    id: doc.id,
-                    username,
-                    ...docData
-                  })
-                }
-              }
-
-              if (changeType === 'removed') {
-                unanswered = unanswered.filter(diagnose => diagnose.id !== doc.id)
-              }
-            })
-          )
-
-          setDiagnoses([...unanswered])
-        })
-    }
-
-    attachQueryListenerForUnansweredDiagnoses()
-  }, [])
-
-  useEffect(() => {
-    const clearScreenIfCurrentDiagnoseWasAnsweredByAnotherOne = () => {
-      if (currentDiagnose) {
-        const queryForCurrentDiagnoseInDiagnoses = diagnoses.filter(diagnose => diagnose.id === currentDiagnose.id)
-        const currentDiagnoseDoesNotExist = queryForCurrentDiagnoseInDiagnoses.length === 0
-        if (currentDiagnoseDoesNotExist) {
-          setCurrentDiagnose(null)
-        }
-      }
-    }
+import './initFirebase'
+import * as firebase from 'firebase'
+import { DiagnoseResponse } from './DiagnoseResponse'
+import { LoginScreen } from './LoginScreen'
+import { Button, Row } from 'react-bootstrap'
 
-    clearScreenIfCurrentDiagnoseWasAnsweredByAnotherOne()
-  }, [diagnoses])
+const App = () => {
+  const [userLoggedIn, setUserLoggedIn] = useState(false)
 
-  const handleSubmit = async (values) => {
-    const dataToUpdate = {
-      answer: values.answer,
-      answeredBy: values.answeredBy,
-      answered: true,
-      updatedAt: firebase.firestore.FieldValue.serverTimestamp()
+  const logOut = async () => {
+    setUserLoggedIn(false)
+    if (firebase.auth().currentUser) {
+      await firebase.auth().signOut()
     }
-
-    await firebase.firestore().collection('diagnoses').doc(currentDiagnose.id).update(dataToUpdate)
-    setDiagnoses(diagnoses.filter(diagnose => diagnose.id !== currentDiagnose.id))
-    setCurrentDiagnose(null)
   }
 
-  const fetchDiagnoseImagesAndSetAsCurrent = async (diagnose) => {
-    const imagesSources = await getDownloadURLFromImages(diagnose.imageReferences)
-    setCurrentDiagnose({
-      ...diagnose,
-      imagesSources
-    })
+  const onUserLogin = () => setUserLoggedIn(true)
+
+  if (userLoggedIn) {
+    return (
+      <>
+        <Row>
+          <Button onClick={logOut}>
+            Log out
+          </Button>
+        </Row>
+        <DiagnoseResponse />
+      </>
+    )
+  } else {
+    return (<LoginScreen onUserLogin={onUserLogin} />)
```

Comment:
There is a convention that when returning one line JSX elements, you do not use parentheses.
Parentheses are used on multiline elements.

[https://github.com/airbnb/javascript/tree/master/react#parentheses](https://github.com/airbnb/javascript/tree/master/react#parentheses)

I personally do not follow all of those conventions but it's just to mention it.

Reply:
We don't use AirBnb's convention, we use StandardJS: https://standardjs.com/

If StandardJS doesn't say anything about the matter, our company-wide convention is to not use parenthesis in this case.